eos/scattering/ee-to-ccbar.cc

```diff
@@ -0,0 +1,673 @@
+#include <array>
+#include <memory>
+
+namespace eos
+{
+
+    template <>
+    struct Implementation<EEToCCBar>
+    {
+        /* Channels and resonances
+        Up to DD*       8  channels 2 resonances
+        Up to D*D*      14 channels 3 resonances
+        Up to 4.8 GeV   24 channels 5 resonances
+         */
+        const static long unsigned nchannels = 24;
+        const static long unsigned nresonances = 5;
+        const static long unsigned order = 0;
+
+        UsedParameter hbar;
+        UsedParameter alpha_em;
+        UsedParameter m_e;
+        UsedParameter m_D0;
+        UsedParameter m_D;
+        UsedParameter m_Dst0;
+        UsedParameter m_Dst;
+        UsedParameter m_Ds;
+        UsedParameter m_Dsst;
+
+        static const std::vector<std::string> resonance_names;
+        static const std::vector<std::string> channel_names;
+
+        //Charmonium masses and sizes
+        std::array<UsedParameter, nresonances> m;
+        std::array<UsedParameter, nresonances> q;
+
+        // Channel-Resonance couplings
+        std::array<std::array<UsedParameter, nchannels>, nresonances> g0;
+
+        // Non-cc contribution to the Rc ratio
+        std::array<std::array<std::array<UsedParameter, nchannels>, nchannels>, order + 1> bkgpol;
+
+        // R_uds
+        UsedParameter Rconstant;
+
+        // Normalization of the exclusive channels
+        UsedParameter exclusive_norm;
+
+        std::shared_ptr<KMatrix<nchannels, nresonances, order>> K;
+
+        using IntermediateResult = EEToCCBar::IntermediateResult;
+        IntermediateResult _intermediate_result;
+
+        template <typename T, T... indices>
+        auto _resonance_masses(const Parameters & p, ParameterUser & u, std::integer_sequence<T, indices...>)
+            -> std::array<UsedParameter, sizeof...(indices)>
+        {
+            if (sizeof...(indices) > resonance_names.size())
+                throw InternalError("The number of requested resonances is larger than the number of resonance names.");
+
+            return std::array<UsedParameter, sizeof...(indices)>
+            {{
+                UsedParameter(p["mass::" + resonance_names[indices]], u)...
+            }};
+        }
+
+        template <typename T, T... indices>
+        auto _resonance_sizes(const Parameters & p, ParameterUser & u, std::integer_sequence<T, indices...>)
+            -> std::array<UsedParameter, sizeof...(indices)>
+        {
+            if (sizeof...(indices) > resonance_names.size())
+                throw InternalError("The number of requested resonances is larger than the number of resonance names.");
+
+            return std::array<UsedParameter, sizeof...(indices)>
+            {{
+                UsedParameter(p["size::" + resonance_names[indices]], u)...
+            }};
+        }
+
+        long unsigned _filter_channel_index(unsigned channel_index)
+        {
+            switch (channel_index)
+            {
+                case 7: // D+D-
```

Comment:
Is this still used?

Reply:
Yes! It filters the channels e.g to have c.c. described by the same parameters